charmcraft/models/manifest.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+"""Model for output charm's manifest.yaml file."""
+import itertools
+import json
+import os
+from typing import Any, Dict, Iterable, List, Literal, Optional
+
+from craft_application import models
+from craft_cli import CraftError
+from pydantic import dataclasses
+from typing_extensions import Self
+
+import charmcraft
+from charmcraft.const import IMAGE_INFO_ENV_VAR
+from charmcraft.linters import CheckResult, CheckType
+from charmcraft.models.charmcraft import Base
+from charmcraft.models.project import Charm
+
+
+@dataclasses.dataclass(frozen=True)
+class Attribute:
+    """An attribute as a linter result."""
+
+    name: str
+    result: str
+
+
+class Manifest(models.BaseMetadata):
+    """A manifest.yaml file for Juju.
+
+    See: https://juju.is/docs/sdk/manifest-yaml
+    """
+
+    charmcraft_version: str = charmcraft.__version__
+    charmcraft_started_at: str
+    bases: Optional[List[Base]]
+    analysis: Dict[Literal["attributes"], List[Attribute]] = {"attributes": []}
```

Comment:
that's a curious structure

Reply:
It is, but when I looked at the spec, the purpose is to allow more analysis output in the manifest in future.